app/controllers/provider/sessions_controller.rb

```diff
@@ -10,8 +10,14 @@ class Provider::SessionsController < FrontendController
 
   def new
     redirect_to provider_admin_dashboard_url if logged_in?
-    @session = Session.new
-    @authentication_providers = published_authentication_providers
+    if domain_account.settings.enforce_sso? && domain_account.sso_authorizations.size == 1
```

Comment:
If you do this: https://github.com/3scale/porta/pull/2155/files#r484300055
Then this line probably should check that the auth provider of the sso_authorization is published before doing the count/size == 1